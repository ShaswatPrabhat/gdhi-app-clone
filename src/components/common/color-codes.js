export default {
  getColorCodes() {
    // Update variables if changed.
    return [
      { score: "5", color: "#01C975", description: "Most Developed" },
      { score: "4", color: "#80E1CC", description: "" },
      { score: "3", color: "#FFE180", description: "" },
      { score: "2", color: "#FFCA82", description: "" },
      { score: "1", color: "#FCAB9C", description: "Least Developed" },
      { score: "NA", color: "#6C757D" },
    ];
  },
};
