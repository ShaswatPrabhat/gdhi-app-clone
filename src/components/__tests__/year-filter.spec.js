import { shallowMount } from "@vue/test-utils";
import { describe, it, expect, beforeEach, vi } from "vitest";
import yearFilter from "../defaultYearSelector/year-filter.vue";
import { EventBus } from "../common/event-bus";
import { i18n } from "../../plugins/i18n";

describe("year-filter", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallowMount(yearFilter, {
      i18n,
      propsData: {
        years: ["Version 1", "2022", "2023"],
        selectedYear: "2022",
        shouldRespectTranslation: true,
      },
    });
  });

  it("should render Years and default Year passed to it", async () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should render Emit Event when a value is filtered", async () => {
    vi.spyOn(EventBus, "$emit");
    const SelectElement = wrapper.find(".year-indicator-select");
    SelectElement.value = "2022";
    SelectElement.trigger("change");
    expect(EventBus.$emit.mock.calls[0][0]).to.equal("year:filtered");
    expect(EventBus.$emit.mock.calls[0][1]).to.equal("2022");
  });
});
