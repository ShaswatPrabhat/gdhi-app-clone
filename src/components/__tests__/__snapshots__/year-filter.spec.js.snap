// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`year-filter > should render Years and default Year passed to it 1`] = `
<div>
  <select
    class="year-indicator-select ltr"
    name="test_select3"
  >
    <option
      value="Version 1"
    >
       Version 1 
    </option>
    <option
      value="2022"
    >
       2022 
    </option>
    <option
      value="2023"
    >
       2023 
    </option>
  </select>
    
</div>
`;
